samza-kafka/src/test/java/org/apache/samza/system/kafka/TestKafkaSystemAdminJava.java

```diff
@@ -63,29 +64,30 @@ public void testCreateChangelogStreamDelegatesToCreateStream() {
     Map<String, ChangelogInfo> changeLogMap = new HashMap<>();
     changeLogMap.put(STREAM, new ChangelogInfo(REP_FACTOR, changeLogProps));
 
-    SystemAdmin admin = Mockito.spy(createSystemAdmin(coordProps, 3, Util.javaMapAsScalaMap(changeLogMap)));
-    StreamSpec spec = new StreamSpec(KafkaSystemAdmin.CHANGELOG_STREAMID(), STREAM, SYSTEM(), PARTITIONS);
-    admin.createChangelogStream(STREAM, PARTITIONS);
+    KafkaSystemAdmin admin = Mockito.spy(createSystemAdmin(coordProps, 3, Util.javaMapAsScalaMap(changeLogMap)));
+    StreamSpec spec = new StreamSpec(StreamSpec.CHANGELOG_STREAM_ID, STREAM, SYSTEM(), PARTITIONS);
+    admin.createStream(spec);
     admin.validateStream(spec);
 
-    ArgumentCaptor<StreamSpec> specCaptor = ArgumentCaptor.forClass(StreamSpec.class);
-    Mockito.verify(admin).createStream(specCaptor.capture());
-
-    StreamSpec internalSpec = specCaptor.getValue();
-    assertTrue(internalSpec instanceof KafkaStreamSpec);  // KafkaStreamSpec is used to carry replication factor
-    assertEquals(KafkaSystemAdmin.CHANGELOG_STREAMID(), internalSpec.getId());
-    assertEquals(SYSTEM(), internalSpec.getSystemName());
-    assertEquals(STREAM, internalSpec.getPhysicalName());
-    assertEquals(REP_FACTOR, ((KafkaStreamSpec) internalSpec).getReplicationFactor());
-    assertEquals(PARTITIONS, internalSpec.getPartitionCount());
-    assertEquals(changeLogProps, ((KafkaStreamSpec) internalSpec).getProperties());
+    Mockito.doAnswer(invocationOnMock -> {
+      StreamSpec internalSpec = (StreamSpec) invocationOnMock.callRealMethod();
+      assertTrue(internalSpec instanceof KafkaStreamSpec);  // KafkaStreamSpec is used to carry replication factor
+      assertEquals(StreamSpec.CHANGELOG_STREAM_ID, internalSpec.getId());
+      assertEquals(SYSTEM(), internalSpec.getSystemName());
+      assertEquals(STREAM, internalSpec.getPhysicalName());
+      assertEquals(REP_FACTOR, ((KafkaStreamSpec) internalSpec).getReplicationFactor());
+      assertEquals(PARTITIONS, internalSpec.getPartitionCount());
+      assertEquals(changeLogProps, ((KafkaStreamSpec) internalSpec).getProperties());
+
+      return internalSpec;
+    }).when(admin).toKafkaSpec(Mockito.any());
   }
 
   @Test
   public void testCreateChangelogStreamDelegatesToCreateStream_specialCharsInTopicName() {
```

Comment:
Not related to your change, but it seems like there should be an equivalent test for coordinator streams with special characters. I thought there was one, but can't find it in this patch. 

I even see it on my local copy.

testCreateCoordinatorStreamDelegatesToCreateStream_specialCharsInTopicName

If it doesn't exist here, would you mind adding it?